### PR TITLE
Add option to write bpf2c output directly to file

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Resolves: #2569

## Description

Add option to write bpf2c output directly to an output file to allow for easier integration with build systems.

## Testing

New tests added.

## Documentation

Help output updated.

### CODE REVIEW

**Comment by tylerretzlaff** [comment]
i didn't review the code for correctness but assuming the code is correct it satisfies the feature request.

**Review by dthaler — APPROVED** [review]

**Comment by dthaler** [comment]
@alanjo can you update tools\bpf2c\templates\{kernel,user}_mode_bpf2c.vcxproj to use the --out option instead of redirecting output?

**Review by dv-msft — DISMISSED** [review]

File: tools/bpf2c/bpf2c.cpp [hunk 1]
```diff
@@ -129,15 +129,33 @@ main(int argc, char** argv)
         auto iter_end = parameters.end();
         std::map<std::string, std::tuple<std::string, std::function<bool()>>> options = {
             {"--sys",
```
Comment (dv-msft) [inline tools/bpf2c/bpf2c.cpp:131]: We might want to extend this string to read `--sys [out_file_name]` for a more helpful visual aid. (same for the other output types).

Comment (dv-msft) [reply tools/bpf2c/bpf2c.cpp:131]: <sigh> can't do it here.  maybe somehow incorporate this in the help text on the next line?

File: tools/bpf2c/bpf2c.cpp [hunk 2]
```diff
@@ -129,15 +129,33 @@ main(int argc, char** argv)
...
-             {"Generate code for a Windows driver",
+             {"Generate code for a Windows driver with optional output file name",
```
Comment (dv-msft) [inline tools/bpf2c/bpf2c.cpp:132]: Might this be a more helpful visual aid?
"Generate code for a Windows driver with optional output file name. (--sys [out_file_name])"

**Comment by dthaler** [comment]
> can you update tools\bpf2c\templates{kernel,user}_mode_bpf2c.vcxproj to use the --out option instead of redirecting output?

@Alan-Jowett still waiting for a response to above comment

**Comment by Alan-Jowett** [comment]
> > can you update tools\bpf2c\templates{kernel,user}_mode_bpf2c.vcxproj to use the --out option instead of redirecting output?
> 
> @Alan-Jowett still waiting for a response to above comment

Should be fixed now.

**Review by dthaler — APPROVED** [review]

**Review by gtrevi — APPROVED** [review]

**Pull request merged**